docs/src/main/asciidoc/extension-authors-guide.adoc

```diff
@@ -874,6 +874,36 @@ As a CDI based runtime, Quarkus extensions often make CDI beans available as par
 However, Quarkus DI solution does not support CDI Portable Extensions.
 Instead, Quarkus extensions can make use of various link:cdi-reference.html[Build Time Extension Points].
 
+=== Extension Health Check
+
+Health checks are provided via the `quarkus-smallrye-health` extension. It provides both liveness and readiness checks capabilities.
+
+When writing an extension, it's beneficial to provide health checks for the extension, that can be automatically included without the developer needing to write their own.
+
+In order to provide a health check, you should do the following:
+
+- Import the `quarkus-smallrye-health` extension as an **optional** dependency in your runtime module so it will not impact the size of the application if
+health check is not included.
+- Create your health check following the link:health-guide.html[Quarkus - MicroProfile Health] guide. We advise providing only
+readiness check for an extension (liveness check is designed to express the fact that an application is up and needs to be lightweight).
+- Import the `quarkus-smallrye-health-spi` library in your deployment module.
+- Add a build step in your deployment module that produces a `HealthBuildItem`.
+- Add a way to disable the extension health check via a config item `quarkus.<extension>.health.enabled` that should be enabled by default.
+
+Following is an example from the Agroal extension that provides a `DataSourceHealthCheck` to validate the readiness of a datasource.
+
+[source%nowrap,java]
+----
+@BuildStep
+HealthBuildItem addHealthCheck(AgroalBuildTimeConfig agroalBuildTimeConfig) {
+    if (agroalBuildTimeConfig.healthEnabled) {
```

Comment:
My main point is that all extensions would need to do that : read the flag and do the if. And when we introduce filtering with wildcard and all, it will be even more conplicated.

Reply:
I add a boolean on the `HealthBuildItem` so that each extension will only needs to pass it's configuration item to it.

At the begining I wanted to retrieve the configuration item in smallrye-health via `ConfigProvider.getConfig()` but it didn't works, I was not able to retrieve the configuration item during the build step ...

Another solution will be to change the way all this is configured from :
```
quarkus.health.extensions.enabled=true # provided by smallrye-health
quarkus.datasource.health.enabled=false # provided by **agroal**
```

To
```
quarkus.health.extensions.enabled=true # provided by smallrye-health
quarkus.health.extensions.datasource.enabled=false # provided by **smallrye-health**
```

Reply:
> I add a boolean on the HealthBuildItem so that each extension will only needs to pass it's configuration item to it.

What do you mean by that? You lost me I think.

Reply:
Ah I think I see now, you mean this https://github.com/quarkusio/quarkus/pull/3671/files#diff-38cd571a6836972129eb3e2eb8528382R901

Reply:
Sounds ok, if reading properties from the SR extension is not an option.